src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -4,20 +4,55 @@
  * Реалізуйте методи для знаходження головної та побічної діагоналі квадратної матриці,
  * максимального та мінімального елементів.
  */
+
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int[] diagonal = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            diagonal[i] = matrix[i][i];
+        }
+        return diagonal;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int[] counterDiagonal = new int[matrix.length];
+        int j = matrix.length - 1;
+        for (int i = 0; i < matrix.length; i++) {
+            counterDiagonal[i] = matrix[i][j];
+            j--;
+        }
+        return counterDiagonal;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        int[] min = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            sortMatrix(matrix[i]);
+            min[i] = matrix[i][0];
+        }
+        sortMatrix(min);
+        return min[0];
     }
 
     public int getMax(int[][] matrix) {
-        return 0;
+        int[] max = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            sortMatrix(matrix[i]);
+            max[i] = matrix[i][matrix.length - 1];
+        }
+        sortMatrix(max);
+        return max[max.length - 1];
+    }
+
+    public int[] sortMatrix(int[] matrix) {
```

Comment:
please, make a solution without sorting the matrix. sorting has a lot of iterations which is not optimal